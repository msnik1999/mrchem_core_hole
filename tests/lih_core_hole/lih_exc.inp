# vim:syntax=sh:

world_prec = 1.0e-3                 # Overall relative precision
world_size = 6                      # Size of simulation box 2^n

MPI {
  numerically_exact = true          # Guarantee identical results in MPI
}

Molecule {
$coords
Li  -3.787412196322  2.239543797522  0.016520727434
H  -6.044538550553  0.576167811227  1.170420150603
$end
}

WaveFunction {
  method = PBE                      # Wave function method (HF or DFT)
}

OrbitalOccupancies {
  occupancies = "0 1 0"
}

SCF {
  run = true
  guess_type = mw                   # Type of initial guess: none, mw, gto
}