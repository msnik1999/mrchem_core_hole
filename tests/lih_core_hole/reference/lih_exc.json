{
    "input": {
      "constants": {
        "N_a": 6.02214076e+23,
        "angstrom2bohrs": 1.8897261246257702,
        "boltzmann_constant": 1.380649e-23,
        "dipmom_au2debye": 2.5417464739297717,
        "e0": 8.8541878128e-12,
        "electron_g_factor": -2.00231930436256,
        "elementary_charge": 1.602176634e-19,
        "fine_structure_constant": 0.0072973525693,
        "hartree2ev": 27.211386245988,
        "hartree2kcalmol": 627.5094740630558,
        "hartree2kjmol": 2625.4996394798254,
        "hartree2simagnetizability": 78.9451185,
        "hartree2wavenumbers": 219474.6313632,
        "light_speed": 137.035999084,
        "meter2bohr": 18897261246.2577
      },
      "geom_opt": {
        "init_step_size": -0.5,
        "max_force_component": 0.005,
        "max_history_length": 10,
        "max_iter": 100,
        "minimal_step_size": 0.01,
        "run": false,
        "subspace_tolerance": 0.001,
        "use_previous_guess": false
      },
      "molecule": {
        "charge": 0,
        "coords": [
          {
            "atom": "li",
            "r_rms": 4.0992133976e-05,
            "xyz": [
              -3.787412196322,
              2.239543797522,
              0.016520727434
            ]
          },
          {
            "atom": "h",
            "r_rms": 2.6569547399e-05,
            "xyz": [
              -6.044538550553,
              0.576167811227,
              1.170420150603
            ]
          }
        ],
        "multiplicity": 1
      },
      "mpi": {
        "bank_size": -1,
        "numerically_exact": true,
        "omp_threads": -1,
        "shared_memory_size": 10000
      },
      "mra": {
        "basis_order": 5,
        "basis_type": "interpolating",
        "boxes": [
          2,
          2,
          2
        ],
        "corner": [
          -1,
          -1,
          -1
        ],
        "max_scale": 20,
        "min_scale": -5
      },
      "printer": {
        "file_name": "imom",
        "print_constants": false,
        "print_level": 0,
        "print_mpi": false,
        "print_prec": 6,
        "print_width": 75
      },
      "rsp_calculations": {},
      "scf_calculation": {
        "fock_operator": {
          "coulomb_operator": {
            "poisson_prec": 0.001,
            "shared_memory": false
          },
          "exchange_operator": {
            "exchange_prec": -1.0,
            "poisson_prec": 0.001
          },
          "kinetic_operator": {
            "derivative": "abgv_55"
          },
          "nuclear_operator": {
            "nuclear_model": "point_like",
            "proj_prec": 0.001,
            "shared_memory": false,
            "smooth_prec": 0.001
          },
          "xc_operator": {
            "shared_memory": false,
            "xc_functional": {
              "cutoff": 0.0,
              "functionals": [
                {
                  "coef": 1.0,
                  "name": "pbe"
                }
              ],
              "spin": false
            }
          }
        },
        "initial_guess": {
          "environment": "None",
          "external_field": "None",
          "file_CUBE_a": "cube_vectors/CUBE_a_vector.json",
          "file_CUBE_b": "cube_vectors/CUBE_b_vector.json",
          "file_CUBE_p": "cube_vectors/CUBE_p_vector.json",
          "file_basis": "initial_guess/mrchem.bas",
          "file_chk": "checkpoint/phi_scf",
          "file_gto_a": "initial_guess/mrchem.moa",
          "file_gto_b": "initial_guess/mrchem.mob",
          "file_gto_p": "initial_guess/mrchem.mop",
          "file_phi_a": "initial_guess/phi_a_scf",
          "file_phi_b": "initial_guess/phi_b_scf",
          "file_phi_p": "initial_guess/phi_p_scf",
          "localize": false,
          "method": "DFT (PBE)",
          "prec": 0.001,
          "relativity": "None",
          "restricted": true,
          "screen": 12.0,
          "type": "mw",
          "zeta": 0
        },
        "occupancies": [
          {
            "occupancy": [
              1.0,
              0.0
            ],
            "orbital": 0
          }
        ],
        "properties": {
          "dipole_moment": {
            "dip-1": {
              "operator": "h_e_dip",
              "precision": 0.001,
              "r_O": [
                0.0,
                0.0,
                0.0
              ]
            }
          }
        },
        "scf_solver": {
          "checkpoint": false,
          "deltascf_method": "none",
          "energy_thrs": -1.0,
          "environment": "None",
          "external_field": "None",
          "file_chk": "checkpoint/phi_scf",
          "final_prec": 0.001,
          "helmholtz_prec": -1.0,
          "kain": 5,
          "localize": false,
          "max_iter": 100,
          "method": "DFT (PBE)",
          "orbital_thrs": 0.01,
          "relativity": "None",
          "rotation": 0,
          "start_prec": 0.001
        }
      },
      "schema_name": "mrchem_input",
      "schema_version": 1
    },
    "output": {
      "properties": {
        "center_of_mass": [
          -4.073619909235116,
          2.0286247074765114,
          0.1628372658138295
        ],
        "charge": 0,
        "dipole_moment": {
          "dip-1": {
            "magnitude": 4.118954272488518,
            "r_O": [
              0.0,
              0.0,
              0.0
            ],
            "vector": [
              -2.8894668688026375,
              2.901848701630694,
              -0.4427636229271319
            ],
            "vector_el": [
              14.517308270715072,
              -4.392950502161806,
              -1.6627459558321673
            ],
            "vector_nuc": [
              -17.40677513951771,
              7.2947992037925005,
              1.2199823329050354
            ]
          }
        },
        "geometry": [
          {
            "symbol": "Li",
            "xyz": [
              -3.787412196322,
              2.239543797522,
              0.016520727434
            ]
          },
          {
            "symbol": "H",
            "xyz": [
              -6.044538550553,
              0.576167811227,
              1.170420150603
            ]
          }
        ],
        "multiplicity": 1,
        "orbital_energies": {
          "energy": [
            -3.1305911583893615,
            -0.5084810858617708
          ],
          "occupation": [
            1.0,
            2.0
          ],
          "spin": [
            "p",
            "p"
          ],
          "sum_occupied": -4.147553330112903
        },
        "orbital_positions": [
          [
            -3.782451258571258,
            2.243194535908221,
            0.014017066859557215
          ],
          [
            -5.367428506071905,
            1.0748779831267923,
            0.8243644444863051
          ]
        ],
        "scf_energy": {
          "E_ee": 2.7687716981108883,
          "E_eext": 0.0,
          "E_el": -6.549448975793505,
          "E_en": -13.346916638485393,
          "E_kin": 5.360549478573394,
          "E_next": 0.0,
          "E_nn": 0.9894513494461199,
          "E_nuc": 0.9894513494461199,
          "E_tot": -5.559997626347385,
          "E_x": 0.0,
          "E_xc": -1.3318535139923944,
          "Er_el": 0.0,
          "Er_nuc": 0.0,
          "Er_tot": 0.0
        }
      },
      "provenance": {
        "creator": "MRChem",
        "mpi_processes": 1,
        "nthreads": 8,
        "routine": "mrchem.x",
        "total_cores": 8,
        "version": "1.2.0-alpha"
      },
      "rsp_calculations": null,
      "scf_calculation": {
        "initial_energy": {
          "E_ee": 2.604960527435693,
          "E_eext": 0.0,
          "E_el": -6.368854542491066,
          "E_en": -12.148694584473262,
          "E_kin": 4.457656896165212,
          "E_next": 0.0,
          "E_nn": 0.9894513494461199,
          "E_nuc": 0.9894513494461199,
          "E_tot": -5.379403193044946,
          "E_x": 0.0,
          "E_xc": -1.2827773816187096,
          "Er_el": 0.0,
          "Er_nuc": 0.0,
          "Er_tot": 0.0
        },
        "scf_solver": {
          "converged": true,
          "cycles": [
            {
              "energy_terms": {
                "E_ee": 2.657412505700292,
                "E_eext": 0.0,
                "E_el": -6.503682179586436,
                "E_en": -12.70604001192246,
                "E_kin": 4.8445850740908565,
                "E_next": 0.0,
                "E_nn": 0.9894513494461199,
                "E_nuc": 0.9894513494461199,
                "E_tot": -5.514230830140316,
                "E_x": 0.0,
                "E_xc": -1.2996397474551258,
                "Er_el": 0.0,
                "Er_nuc": 0.0,
                "Er_tot": 0.0
              },
              "energy_total": -5.514230830140316,
              "energy_update": 0.13482763709536982,
              "mo_residual": 0.1619159050534102,
              "wall_time": 1.109493417
            },
            {
              "energy_terms": {
                "E_ee": 2.7150725377894456,
                "E_eext": 0.0,
                "E_el": -6.534726392151677,
                "E_en": -13.049683214725125,
                "E_kin": 5.119816463481179,
                "E_next": 0.0,
                "E_nn": 0.9894513494461199,
                "E_nuc": 0.9894513494461199,
                "E_tot": -5.5452750427055575,
                "E_x": 0.0,
                "E_xc": -1.319932178697178,
                "Er_el": 0.0,
                "Er_nuc": 0.0,
                "Er_tot": 0.0
              },
              "energy_total": -5.5452750427055575,
              "energy_update": 0.03104421256524148,
              "mo_residual": 0.08568066839928926,
              "wall_time": 0.900934792
            },
            {
              "energy_terms": {
                "E_ee": 2.7458103920834023,
                "E_eext": 0.0,
                "E_el": -6.544362667072955,
                "E_en": -13.220810481372826,
                "E_kin": 5.260096903454098,
                "E_next": 0.0,
                "E_nn": 0.9894513494461199,
                "E_nuc": 0.9894513494461199,
                "E_tot": -5.554911317626835,
                "E_x": 0.0,
                "E_xc": -1.3294594812376288,
                "Er_el": 0.0,
                "Er_nuc": 0.0,
                "Er_tot": 0.0
              },
              "energy_total": -5.554911317626835,
              "energy_update": 0.009636274921277277,
              "mo_residual": 0.04805872698680878,
              "wall_time": 0.900082
            },
            {
              "energy_terms": {
                "E_ee": 2.7761304952974597,
                "E_eext": 0.0,
                "E_el": -6.549301120024381,
                "E_en": -13.378428567957304,
                "E_kin": 5.388554491517023,
                "E_next": 0.0,
                "E_nn": 0.9894513494461199,
                "E_nuc": 0.9894513494461199,
                "E_tot": -5.5598497705782615,
                "E_x": 0.0,
                "E_xc": -1.335557538881559,
                "Er_el": 0.0,
                "Er_nuc": 0.0,
                "Er_tot": 0.0
              },
              "energy_total": -5.5598497705782615,
              "energy_update": 0.004938452951426697,
              "mo_residual": 0.06090416878624683,
              "wall_time": 0.942698541
            },
            {
              "energy_terms": {
                "E_ee": 2.7687716981108883,
                "E_eext": 0.0,
                "E_el": -6.549448975793505,
                "E_en": -13.346916638485393,
                "E_kin": 5.360549478573394,
                "E_next": 0.0,
                "E_nn": 0.9894513494461199,
                "E_nuc": 0.9894513494461199,
                "E_tot": -5.559997626347385,
                "E_x": 0.0,
                "E_xc": -1.3318535139923944,
                "Er_el": 0.0,
                "Er_nuc": 0.0,
                "Er_tot": 0.0
              },
              "energy_total": -5.559997626347385,
              "energy_update": 0.0001478557691232396,
              "mo_residual": 0.006022571910403009,
              "wall_time": 0.950230208
            }
          ],
          "wall_time": 4.805341708
        },
        "success": true
      },
      "schema_name": "mrchem_output",
      "schema_version": 1,
      "success": true
    }
  }