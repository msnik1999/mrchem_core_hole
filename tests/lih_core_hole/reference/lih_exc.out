***************************************************************************
***                                                                     ***
***                                                                     ***
***          __  __ ____   ____ _                                       ***
***         |  \/  |  _ \ / ___| |__   ___ _ __ ___                     ***
***         | |\/| | |_) | |   | '_ \ / _ \ '_ ` _ \                    ***
***         | |  | |  _ <| |___| | | |  __/ | | | | |                   ***
***         |_|  |_|_| \_\\____|_| |_|\___|_| |_| |_|                   ***
***                                                                     ***
***         VERSION            1.2.0-alpha                              ***
***                                                                     ***
***         Git branch         mom_merged                               ***
***         Git commit hash    a1a2e5991d8adbd8494b-dirty               ***
***         Git commit author  Niklas GÃ¶llmann                         ***
***         Git commit date    Fri May 10 13:22:30 2024 +0200           ***
***                                                                     ***
***         Contact: luca.frediani@uit.no                               ***
***                                                                     ***
***         Radovan Bast            Magnar Bjorgve                      ***
***         Roberto Di Remigio      Antoine Durdek                      ***
***         Luca Frediani           Gabriel Gerez                       ***
***         Stig Rune Jensen        Jonas Juselius                      ***
***         Rune Monstad            Peter Wind                          ***
***                                                                     ***
***************************************************************************

---------------------------------------------------------------------------

 MPI processes           :       (no bank)                               1
 OpenMP threads          :                                               8
 Total cores             :                                               8

---------------------------------------------------------------------------

XCFun DFT library Copyright 2009-2020 Ulf Ekstrom and contributors.
See http://dftlibs.org/xcfun/ for more information.

This is free software; see the source code for copying conditions.
There is ABSOLUTELY NO WARRANTY; not even for MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE. For details see the documentation.
Scientific users of this library should cite
U. Ekstrom, L. Visscher, R. Bast, A. J. Thorvaldsen and K. Ruud;
J.Chem.Theor.Comp. 2010, DOI: 10.1021/ct100117s

---------------------------------------------------------------------------

 MRCPP version         : 1.6.0-alpha
 Git branch            : mom
 Git commit hash       : 9ebc1da6290a65a252e6
 Git commit author     : Laura Ratcliff
 Git commit date       : Wed Jun 14 14:11:39 2023 +0200

 Linear algebra        : EIGEN v3.4.0
 Parallelization       : OpenMP (8 threads)

---------------------------------------------------------------------------



===========================================================================
                         MultiResolution Analysis
---------------------------------------------------------------------------
 polynomial order      : 5
 polynomial type       : Interpolating
---------------------------------------------------------------------------
 total boxes           : 8
 boxes                 : [          2           2           2 ]
 unit lengths          : [   32.00000    32.00000    32.00000 ]
 scaling factor        : [    1.00000     1.00000     1.00000 ]
 lower bounds          : [  -32.00000   -32.00000   -32.00000 ]
 upper bounds          : [   32.00000    32.00000    32.00000 ]
 total length          : [   64.00000    64.00000    64.00000 ]
===========================================================================



***************************************************************************
***                                                                     ***
***                        Initializing Molecule                        ***
***                                                                     ***
***************************************************************************


===========================================================================
                                 Molecule
---------------------------------------------------------------------------
 Charge                  :                                               0
 Multiplicity            :                                               1
---------------------------------------------------------------------------
    N    Atom            :               x               y               z
---------------------------------------------------------------------------
    0      Li            :       -3.787412        2.239544        0.016521
    1       H            :       -6.044539        0.576168        1.170420
---------------------------------------------------------------------------
 Center of mass          :       -4.073620        2.028625        0.162837
===========================================================================



***************************************************************************
***                                                                     ***
***                Computing Initial Guess Wavefunction                 ***
***                                                                     ***
***************************************************************************


Warning: guess_orbitals(), line 405: Occupancies for alpha and beta orbitals are added together for restricted calculations
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation             : Compute initial orbitals
 Method                  : Project MW molecular orbitals
 Precision               : 1.00000e-03
 Restricted              : True
 MO file                 : initial_guess/phi_p_scf
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


===========================================================================
                            Molecular Orbitals
---------------------------------------------------------------------------
 Alpha electrons         :                                           1.500
 Beta electrons          :                                           1.500
 Total electrons         :                                           3.000
---------------------------------------------------------------------------
    n     Occ Spin       :                                            Norm
---------------------------------------------------------------------------
    0    1.00    p       :                              9.999999504019e-01
    1    2.00    p       :                              9.999999980940e-01
===========================================================================


~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation             : Compute initial energy
 Method                  : DFT (PBE)
 Relativity              : None
 Environment             : None
 External fields         : None
 Precision               : 1.00000e-03
 Localization            : Off
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


===========================================================================
                        Molecular Energy (initial)
---------------------------------------------------------------------------
 Kinetic energy          :            (au)                  4.457656896165
 E-N energy              :            (au)                -12.148694584473
 Coulomb energy          :            (au)                  2.604960527436
 Exchange energy         :            (au)                  0.000000000000
 X-C energy              :            (au)                 -1.282777381619
 N-N energy              :            (au)                  0.989451349446
---------------------------------------------------------------------------
 Electronic energy       :            (au)                 -6.368854542491
 Nuclear energy          :            (au)                  0.989451349446
---------------------------------------------------------------------------
 Total energy            :            (au)             -5.379403193045e+00
                         :      (kcal/mol)             -3.375626468441e+03
                         :        (kJ/mol)             -1.412362114396e+04
                         :            (eV)             -1.463810180588e+02
===========================================================================


===========================================================================
                        Orbital Energies (initial)
---------------------------------------------------------------------------
    n     Occ Spin       :                                         Epsilon
---------------------------------------------------------------------------
    0    1.00    p       :            (au)                 -3.302644782934
    1    2.00    p       :            (au)                 -0.407658022451
---------------------------------------------------------------------------
 Sum occupied            :            (au)                 -4.117960827836
===========================================================================



***************************************************************************
***                                                                     ***
***                 Computing Ground State Wavefunction                 ***
***                                                                     ***
***************************************************************************


~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation             : Optimize ground state orbitals
 Method                  : DFT (PBE)
 Relativity              : None
 Environment             : None
 External fields         : None
 Checkpointing           : Off
 Max iterations          : 100
 KAIN solver             : 5
 Localization            : Off
 Diagonalization         : First two iterations
 Start precision         : 1.00000e-03
 Final precision         : 1.00000e-03
 Helmholtz precision     : Dynamic
 Energy threshold        : Off
 Orbital threshold       : 1.00000e-02
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


===========================================================================
 Iter           MO residual             Total energy                Update
---------------------------------------------------------------------------
    0          1.414214e+00          -5.379403193045         -5.379403e+00
    1          1.619159e-01          -5.514230830140         -1.348276e-01
    2          8.568067e-02          -5.545275042706         -3.104421e-02
    3          4.805873e-02          -5.554911317627         -9.636275e-03
    4          6.090417e-02          -5.559849770578         -4.938453e-03
    5          6.022572e-03          -5.559997626347         -1.478558e-04
---------------------------------------------------------------------------
                      SCF converged in 5 iterations!
===========================================================================



***************************************************************************
***                                                                     ***
***                    Printing Molecular Properties                    ***
***                                                                     ***
***************************************************************************


===========================================================================
                                 Molecule
---------------------------------------------------------------------------
 Charge                  :                                               0
 Multiplicity            :                                               1
---------------------------------------------------------------------------
    N    Atom            :               x               y               z
---------------------------------------------------------------------------
    0      Li            :       -3.787412        2.239544        0.016521
    1       H            :       -6.044539        0.576168        1.170420
---------------------------------------------------------------------------
 Center of mass          :       -4.073620        2.028625        0.162837
===========================================================================


===========================================================================
                         Molecular Energy (final)
---------------------------------------------------------------------------
 Kinetic energy          :            (au)                  5.360549478573
 E-N energy              :            (au)                -13.346916638485
 Coulomb energy          :            (au)                  2.768771698111
 Exchange energy         :            (au)                  0.000000000000
 X-C energy              :            (au)                 -1.331853513992
 N-N energy              :            (au)                  0.989451349446
---------------------------------------------------------------------------
 Electronic energy       :            (au)                 -6.549448975794
 Nuclear energy          :            (au)                  0.989451349446
---------------------------------------------------------------------------
 Total energy            :            (au)             -5.559997626347e+00
                         :      (kcal/mol)             -3.488951186301e+03
                         :        (kJ/mol)             -1.459777176348e+04
                         :            (eV)             -1.512952429373e+02
===========================================================================


===========================================================================
                         Orbital Energies (final)
---------------------------------------------------------------------------
    n     Occ Spin       :                                         Epsilon
---------------------------------------------------------------------------
    0    1.00    p       :            (au)                 -3.130591158389
    1    2.00    p       :            (au)                 -0.508481085862
---------------------------------------------------------------------------
 Sum occupied            :            (au)                 -4.147553330113
===========================================================================


===========================================================================
                             Orbital Positions
---------------------------------------------------------------------------
Index                    :               x               y               z
---------------------------------------------------------------------------
    0                    :       -3.782451        2.243195        0.014017
    1                    :       -5.367429        1.074878        0.824364
===========================================================================


===========================================================================
                           Dipole Moment (dip-1)
---------------------------------------------------------------------------
 r_O                     :        0.000000        0.000000        0.000000
---------------------------------------------------------------------------
 Electronic vector       :       14.517308       -4.392951       -1.662746
 Magnitude               :            (au)                       15.258276
                         :         (Debye)                       38.782668
---------------------------------------------------------------------------
 Nuclear vector          :      -17.406775        7.294799        1.219982
 Magnitude               :            (au)                       18.912913
                         :         (Debye)                       48.071830
---------------------------------------------------------------------------
 Total vector            :       -2.889467        2.901849       -0.442764
 Magnitude               :            (au)                        4.118954
                         :         (Debye)                       10.469337
===========================================================================




***************************************************************************
***                                                                     ***
***                            Exiting MRChem                           ***
***                                                                     ***
***                       Wall time :  0h  0m  5s                       ***
***                                                                     ***
***************************************************************************